### PR TITLE
IGNITE-21933 Fix TxStateStorage#leaseStartTime possible inconsistency with partition storage

### PR DESCRIPTION
**Pull request by denis-chudov**
https://issues.apache.org/jira/browse/IGNITE-21933

### CODE REVIEW

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMetaStorage.java [hunk 1]
```diff
@@ -54,6 +54,11 @@ public class RocksDbMetaStorage {
      */
     public static final byte[] INDEX_ROW_ID_PREFIX = {2};
 
+    /**
+     * Prefix to store lease start time. Key format is {@code [prefix, tableId, partitionId]} in BE.
+     */
+    static final byte[] LEASE_PREFIX = {3};
```
Comment (ibessonov) [inline modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMetaStorage.java:60]: Do we clean it on partition/table destruction? I think you forgot it

Comment (denis-chudov) [reply modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMetaStorage.java:60]: fixed

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java [hunk 2]
```diff
@@ -1035,6 +1032,7 @@ public long leaseStartTime() {
     void destroyData(WriteBatch writeBatch) throws RocksDBException {
         writeBatch.delete(meta, lastAppliedIndexAndTermKey);
         writeBatch.delete(meta, lastGroupConfigKey);
+        writeBatch.delete(meta, leaseKey);
```
Comment (ibessonov) [inline modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java:1035]: Please fix `org.apache.ignite.internal.storage.rocksdb.instance.SharedRocksDbInstance#destroyTable` as well. Maybe there are other places, please be careful and ensure that you covered everything

Comment (denis-chudov) [reply modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java:1035]: Thanks, fixed. I see no other usages of prefixes

**Review by ibessonov — APPROVED** [review]

**Pull request merged**